### PR TITLE
Generic Trigger System & Ancient Temple Changes Pt1

### PR DESCRIPTION
**Pull request by AtnerNT**
Was originally all gonna be one PR, but Gilles wanted to do some stuff with the things already finished, so merging what is already working 
Whats In Progress but not in this PR:
-> New Ancient Temple map complete with a variety of traps, puzzles and new loot
-> Sprite work for logic wall switches (Currently using logic gate sprite)
-> Small bug fixes for logic items such as spike traps having a desynced enable/disable animation

### Changelog:

CL: [New] Adds several new vinyls and songs by M0N7Y5
CL: [New] Adds mappable logic gates, timers and clearance providers to increase mapping capabilities
CL: [New] Adds pressure plates, timers and logic switches
CL: [New] Adds spike traps and dart traps
CL: [New] Adds two new crate types, "Stone Chests" and its vined variant for more thematic loot in awaysites
CL: [New] Adds new floor tile "Sandstone Floor"

CL: [Fix] Fixed projectiles that dont spawn decals forming NREs and breaking, fixing certain bugs with arrow variants
CL: [Improvement] The research laser projector will now display dismantling hovertips even if not already holding the correct tool

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
tests are failing

**Comment by AtnerNT** [comment]
Ay tests passing now

File: UnityProject/Assets/Scripts/Objects/Traps/LogicGate.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+			return null;
+		}
+
+		public List<TextColor> InteractionsStrings()
+		{
+			List<TextColor> interactions = new List<TextColor>();
+
+			TextColor text = new TextColor
+			{
+				Text = $"Interact to change active input for multitool interaction.\nCurrently interacting with: {RetrieveTrigger().gameObject.name}",
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Objects/Traps/LogicGate.cs]: Wouldn't this be null if there's no trigger? add a check to make sure that we aren't trying to access something that doesn't exist

File: UnityProject/Assets/Scripts/Objects/Traps/LogicTimer.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+				{
+					Text = $"The timer is currently active",
+					Color = IntentColors.Help
+				};
+			}
+			else
+			{
+				text = new TextColor
+				{
+					Text = $"The timer is currently inactive",
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Objects/Traps/LogicTimer.cs]: Interaction strings are meant for player interactions; not information.
Move this to hover tip.

**Review by MaxIsJoe — CHANGES_REQUESTED** [review]

**Comment by AtnerNT** [comment]
reviews applied

**Review by MaxIsJoe — APPROVED** [review]

**Pull request merged**